oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticConnection.java

```diff
@@ -108,11 +121,25 @@ public RestHighLevelClient getClient() {
                         Header[] headers = new Header[]{new BasicHeader("Authorization", "ApiKey " + apiKeyAuth)};
                         builder.setDefaultHeaders(headers);
                     }
-                    client = new RestHighLevelClient(builder);
+                    hlClient = new RestHighLevelClient(builder);
```

Comment:
Let's say there is an exception or error (RuntimeException, OutOfMemoryError,...) after setting the hlClient but before setting the esClient. In this case, we end up having hlClient != null and esClient == null, which is a weird state... 

To protect against that, the only 100% save way I would know is encapsulate esClient + hlClient in a separate object (a "ClientPair" class) and assign that... but it would be overly complicated I think. An alternative is to do the following (untested):

    RestHighLevelClient rc = new RestHighLevelClient(builder);
    ElasticsearchClient ec = new ElasticsearchClient(new RestClientTransport(
            	        rc.getLowLevelClient(), new JacksonJsonpMapper()));
    // "atomically" assign to the fields
    hlClient = rc;
    esClient = ec;

Reply:
I don't see this case as possible. I trust you and I assume it is a protective strategy to implement the code. I will try to understand and follow.
What I would like to assess beforehand is whether we are even contemplating the option of merging the code before it the client migration (recoding) has been completed. If so, yes we should be protective and I should put more attention on it.

Reply:
the clients are now atomically returned in a Tuple.